openstack/blockstorage/extensions/quotasets/results.go

```diff
@@ -34,6 +36,42 @@ type QuotaSet struct {
 
 	// Groups is the number of groups that are allowed for each project.
 	Groups int `json:"groups,omitempty"`
+
+	// Extra is a collection of miscellaneous key/values used to set
+	// quota per volume_type
+	Extra map[string]interface{} `json:"-"`
+}
+
+// UnmarshalJSON is used on QuotaSet to unmarshal extra keys that are
+// used for volume_type quota
+func (r *QuotaSet) UnmarshalJSON(b []byte) error {
+	type tmp QuotaSet
+	var s struct {
+		tmp
+		Extra map[string]interface{} `json:"extra"`
+	}
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	*r = QuotaSet(s.tmp)
+
+	// Collect other fields and bundle them into Extra
+	// but only if a field titled "extra" wasn't sent.
+	if s.Extra != nil {
```

Comment:
This is specific to the Keystone/Identity API and can be removed here. The body of the else clause should only be used.

Reply:
done